lib/text/ttml_text_parser.js

```diff
@@ -239,17 +239,17 @@ shaka.text.TtmlTextParser = class {
       return cue;
     }
 
-    // Disregard empty elements:
-    // TTML allows for empty elements like <div></div>.
-    // If cueElement has neither time attributes, nor
-    // non-whitespace text, don't try to make a cue out of it.
-    if (
-      /^[\s\n]*$/.test(cueElement.textContent) ||
-      (
-        !isNested &&
+    const isTextContentEmpty = /^[\s\n]*$/.test(cueElement.textContent);
+    const hasNoTimeAttributes = cueElement.nodeType == Node.ELEMENT_NODE &&
         !cueElement.hasAttribute('begin') &&
-        !cueElement.hasAttribute('end')
-      )
+        !cueElement.hasAttribute('end');
+
+    if (
```

Comment:
Please add a more detailed comment on this case. Why are we letting through the elements with time stamps? Why are we ignoring the ones without?

Reply:
Sure thing :)